src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/MarshalAsAttributeParser.cs

```diff
@@ -98,6 +97,18 @@ UseSiteAttributeData IUseSiteAttributeParser.ParseAttribute(AttributeData attrib
                 }
             }
 
+            // We'll support the UnmanagedType.Interface option, but we'll explicitly
+            // exclude ComImport types as they will not work as expected
+            // unless they are migrated to [GeneratedComInterface].
+            if (unmanagedType == UnmanagedType.Interface)
+            {
+                if (type is INamedTypeSymbol { IsComImport: true })
```

Comment:
Do we need a test for this? Not sure how easy it would be but I appreciate the comment here regardless.